### ISSUE TITLE
Consider moving full-text search to a Lambda function

### ISSUE THREAD

**Issue by peterbe** [mdn/yari#1226]
Our current (Sept 2020) plan is that we, as part of the deployer, send in all rendered HTML and titles and stuff to an Elasticsearch database, from the GitHub Action Production Build workflow. Then, we'd use our Python code in Kuma (django) transform the v1 API search parameters into Elasticsearch queries and back again as JSON which Yari's front-end would read. 

Another option is **to bypass Django entirely**. In the deployer we bulk submit all the content to our Elasticsearch and then we write a Lambda function that can have the Elasticsearch URL as its only secret. Now, we just need to write a Lambda function to do convert the search query string parameters into a Elasticsearch DSL query and then return it as JSON.

**Comment by peterbe** [comment]
Oh. One thing that occurred to me is that Yari1 will use the Django Kuma implementation to *render* the full-text search results. So, in that regard, we're pretty close to that already.

**Comment by escattone** [comment]
@peterbe I think this is a good idea, but definitely not something as part of the Yari1 launch, since we can continue to use Django. I think we have to avoid adding any new work to the Yari1 launch unless it's absolutely unavoidable.

**Comment by peterbe** [comment]
Ryan and I have discussed this quite a lot. We decided it's easier to move it to Kuma. 
In fact, I have a fully working prototype that already does exactly that. 
The key trigger point was the realization that once we're no longer a Wiki, we can bulldozer through all and any ES related code in the Kuma code base because we no longer need to worry about hooking up to the Wiki ORM to send snapshots to ES.
